krossbow-websocket-core/src/nativeDarwinMain/kotlin/org/hildan/krossbow/websocket/darwin/DarwinWebSocketClient.kt

```diff
@@ -113,7 +113,7 @@ private class IosWebSocketListener(
             return
         }
 
-        incomingFrames.close(didCompleteWithError.toIosWebSocketException())
+        incomingFrames.close(didCompleteWithError.toIosWebSocketException(task.response))
```

Comment:
Will there ever be a `NSHTTPURLResponse` when we're not in the handshake? The `if (isConnecting)` should deal with handshake errors, so this code path should only be reached when an error occurs while the web socket is already opened - and in that case we should never have a meaningful HTTP error code.

Worse, it could be that we get an HTTP 200 from the initial successful handshake - and that would be quite weird in the exception message.

Reply:
yeah I'm not sure actually, might be safer to only use it in the `if (isConnecting)` block then

Reply:
Makes sense, thanks! Actually there is already a `WebSocketConnectionException` which is a special case of `WebSocketException` for the handshake. I think I will add the HTTP status code in that one as well, so all implementations can have it, and we will even be able to extend it directly instead of throwing it with the darwin one as cause.

Reply:
oh yeah that sounds like a great idea